src/stdatamodels/model_base.py

```diff
@@ -138,8 +146,12 @@ def __init__(self, init=None, schema=None, memmap=False,
         if strict_validation is None:
             strict_validation = get_envar_as_boolean("STRICT_VALIDATION",
                                                           False)
+        if validate_on_assignment is None:
+            validate_on_assignment = get_envar_as_boolean("VALIDATE_ON_ASSIGNMENT",
+                                                          True)
```

Comment:
Since the default is `validate_on_assignment=True`, the env var will never have an effect unless explicitly set to `None`.  I.e.:

```python
DataModel(validate_no_assignment=None)
```

So it might be better to have the default be `None`.

Reply:
I changed the default to `None` in order to use environment variables and everything passed pytest.